coveo-settings/coveo_settings/validation.py

```diff
@@ -0,0 +1,15 @@
+from typing import Optional, Sequence
+
+
+class InSequence:
+    def __init__(self, *args: str, condition: Optional[Sequence[str]] = None) -> None:
+        if condition is None:
+            self._condition = tuple(args)
+        else:
+            self._condition = tuple(condition)
```

Comment:
I don't think the `condition` adds much to the case, except maybe confusion. I would keep only the *args.

Reply:
Yes you are right, I first added **condition** argument and then though to myself : wouldn't it be much easier to use `*args` :stuck_out_tongue:.